pkg/tfvars/internal/cache/cache.go

```diff
@@ -193,9 +199,90 @@ func DownloadFile(baseURL string, dataType string) (string, error) {
 }
 
 // DownloadImageFile is a helper function that obtains an image file from a given URL,
-// puts it in the cache and returns the local file path.
+// puts it in the cache and returns the local file path.  If the file is compressed
+// by a known compressor, the file is uncompressed prior to being returned.
 func DownloadImageFile(baseURL string) (string, error) {
 	logrus.Infof("Obtaining RHCOS image file from '%v'", baseURL)
 
 	return DownloadFile(baseURL, imageDataType)
 }
+
+func detectContentType(file string) (string, error) {
+	logrus.Debugf("Detecting content type for %s", file)
+	fd, err := os.Open(file)
+	if err != nil {
+		logrus.Errorf("detectContentType unable to open file %s: %v", file, err)
+		return "", err
+	}
+	defer fd.Close()
+
+	// See http://golang.org/pkg/net/http/#DetectContentType for why we use 512
+	buf := make([]byte, 512)
+	_, err = fd.Read(buf)
+	if err != nil {
+		logrus.Errorf("detectContentType unable to read file %s: %v", file, err)
+		return "", err
+	}
+	filetype := http.DetectContentType(buf)
+	logrus.Debugf("detectContentType %s is %s", file, filetype)
+	return filetype, nil
+}
+
+func decompressFileGzip(src, dest string) error {
+	logrus.Debugf("decompressFileGzip %s %s", src, dest)
+	gzipfile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+
+	reader, err := gzip.NewReader(gzipfile)
+	defer reader.Close()
+	if err != nil {
+		return err
+	}
+
+	writer, err := os.Create(dest)
+	defer writer.Close()
+	defer os.Remove(dest)
+	if err != nil {
+		return err
+	}
+
+	if _, err = io.Copy(writer, reader); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// decompressFile decompresses data in the cache
+func decompressFile(src, dest string) error {
+	filetype, err := detectContentType(src)
+	if err != nil {
+		return err
+	}
+	if decompressor, exists := compressorMap[filetype]; exists {
+		if src == dest {
+			tmpSrc := src + ".tmp"
+			err = os.Rename(src, tmpSrc)
+			if err != nil {
+				return err
+			}
+			src = tmpSrc
+		}
+		uncompressTmp := src + "_compressed"
```

Comment:
The naming here is a bit confusing, foo.tmp_compressed but it's actually the uncompressed file?

If we can keep track of a reasonable file extension in the compressorMap it might be nice to just rename to e.g `foo.gz`, then uncompress to `foo.tmp` or something but if not then perhaps this should be `tmp_uncompressed`?

Reply:
Better yet perhaps would be to have the decompress function return the name of the temp file it creates.